src/routing_table/mod.rs

```diff
@@ -694,4 +701,132 @@ mod tests {
         assert!(table.is_empty());
         assert_eq!(table.iter().count(), 0);
     }
+
+    #[test]
+    fn test_routing_groups() {
+        // Use replicable random numbers to initialse a table:
+        use rand::{Rng, SeedableRng, XorShiftRng};
+        let mut rng: XorShiftRng = SeedableRng::from_seed([1315, 30, 61894, 315]);
+        let our_name = rng.next_u32();
+        let mut table = RoutingTable::new(our_name, 8);
+        let mut unknown_distant_name = None;
+
+        for _ in 0..1000 {
```

Comment:
How long does this test take? Let's just make sure to not hold up the CI tests even more. But probably it's negligible, compared to the mock crust tests?